private/model/api/behavior.go

```diff
@@ -0,0 +1,320 @@
+	a.AddSDKImport("aws/corehandlers")
+	a.AddSDKImport("aws/request")
+	a.AddSDKImport("private/protocol")
+	a.AddSDKImport("internal/sdktesting")
+
+	a.AddImport(a.ImportPath())
+
+	behaviorTests := struct {
+		API *API
+		BehaviorTestSuite
+	}{
+		API:               a,
+		BehaviorTestSuite: a.BehaviorTests,
+	}
+
+	if err := behaviorTestTmpl.Execute(w, behaviorTests); err != nil {
+		panic(fmt.Sprintf("failed to create behavior tests, %v", err))
+	}
+
+	ignoreImports := `
+	var _ *time.Time
+	var _ = protocol.ParseTime
+	var _ = strings.NewReader
+	var _ = json.Marshal
+	`
+
+	return a.importsGoCode() + ignoreImports + w.String()
+}
+
+//Generates assertions
+func (c Case) GenerateAssertions(op *Operation) string {
+	var val string = "//Assertions start here"
+	for _, assertion := range c.Expect {
+		for assertionName, assertionContext := range assertion {
+			val += fmt.Sprintf("\nawstesting.Assert")
+
+			switch assertionName {
+			case "requestMethodEquals":
+				val += fmt.Sprintf("RequestMethodEquals(t, %q, req.HTTPRequest.Method)", assertionContext)
+			case "requestUrlMatches":
+				val += fmt.Sprintf("RequestURLMatches(t, %q, req.HTTPRequest.URL.String())", assertionContext)
+			case "requestUrlPathMatches":
+				val += fmt.Sprintf("RequestURLPathMatches(t, %q, req.HTTPRequest.URL.EscapedPath())", assertionContext)
+			case "requestUrlQueryMatches":
+				val += fmt.Sprintf("RequestURLQueryMatches(t, %q, req)", assertionContext)
+			case "requestHeadersMatch":
+				val += fmt.Sprintf("RequestHeadersMatch(t, %#v, req)", assertionContext)
+			case "requestBodyEqualsBytes":
+				val += fmt.Sprintf("RequestBodyEqualsBytes(t, %q, req)", assertionContext)
+			case "requestBodyEqualsJson":
+				val += fmt.Sprintf("RequestBodyEqualsJSON(t, %#v, req)", assertionContext)
+			case "requestBodyMatchesXml":
+				val += fmt.Sprintf("RequestBodyMatchesXML(t, %q, req, %v)", assertionContext, c.Request.EmptyShapeBuilder(&op.InputRef))
+			case "requestBodyEqualsString":
+				val += fmt.Sprintf("RequestBodyEqualsString(t, %q, req)", assertionContext)
+			case "requestIdEquals":
+				val += fmt.Sprintf("RequestIDEquals(t, %q, req.RequestID)", assertionContext)
+			case "responseDataEquals":
+				val += fmt.Sprintf("ResponseDataEquals(t, %v, resp)", c.BuildOutputShape(&op.OutputRef))
+			case "responseErrorDataEquals":
+				val += fmt.Sprintf("ResponseErrorDataEquals(t, %#v, err)", assertionContext)
+			case "responseErrorIsKindOf":
+				val += fmt.Sprintf("ResponseErrorIsKindOf(t, %q, err)", assertionContext)
+			case "responseErrorMessageEquals":
+				val += fmt.Sprintf("ResponseErrorMessageEquals(t, %q, err)", assertionContext)
+			case "responseErrorRequestIdEquals":
+				val += fmt.Sprintf("ResponseErrorRequestIDEquals(t, %q, err)", assertionContext)
+			default:
+				panic("Invalid assertion key")
+			}
+		}
+	}
+	return val
+}
+
+// Returns a value to set Credentials
+func (c Case) ConfigurationString(T Tests) string {
```

Comment:
`Tests` struct probably should be renamed to something like `BehaviorTests`. `Tests` is too generic.

Reply:
Changed to `BehaviorTest` because the name `BehaviorTests` is used to initialize `BehaviorTestSuite`  in `api.go` (To be consistent with smoke tests initialization)